src/base/Collection.i.hh

```diff
@@ -37,12 +37,12 @@ Collection<T, W, M, I>::Collection(Collection<T, W2, M2, I>& other)
 
 //---------------------------------------------------------------------------//
 /*!
- * Assign from another collection in the same memory space.
+ * Assign from another collection.
  */
 template<class T, Ownership W, MemSpace M, class I>
-template<Ownership W2>
+template<Ownership W2, MemSpace M2>
 Collection<T, W, M, I>&
-Collection<T, W, M, I>::operator=(const Collection<T, W2, M, I>& other)
+Collection<T, W, M, I>::operator=(const Collection<T, W2, M2, I>& other)
```

Comment:
I lost track (and hence it probably would be helpful to document in the code) how the assignment to/from the wrong memory space is prevented and/or where the migration from one to the other is handled.

Reply:
It's in the `CollectionAssigner` -- I'll add some comments. The reason I had to make this change was because the previous implementation was relying on the implicit constructor (which *did* allow for memory space "conversion", even though it did static checking to make sure it was safe).